src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -555,13 +556,35 @@ def _get_qualified_name(self, node: ast.AST, *, joiner="."):
 
         return result
 
-    @staticmethod
-    def _sort_blocks_key(comp_block):
-        start = comp_block.interval.start
-        assert isinstance(start.level, gt_ir.LevelMarker)
-        key = 0 if start.level == gt_ir.LevelMarker.START else 100000
-        key += start.offset
-        return key
+    def _are_blocks_sorted(self, compute_blocks: List[gt_ir.ComputationBlock]):
+        def sort_blocks_key(comp_block):
+            start = comp_block.interval.start
+            assert isinstance(start.level, gt_ir.LevelMarker)
+            key = 0 if start.level == gt_ir.LevelMarker.START else 100000
+            key += start.offset
+            return key
+
+        if len(compute_blocks) < 1:
+            return True
+
+        # validate invariant
+        assert all(
+            comp_block.iteration_order == compute_blocks[0].iteration_order
+            for comp_block in compute_blocks
+        )
+
+        # extract iteration order
+        iteration_order = compute_blocks[0].iteration_order
+
+        # sort blocks
+        compute_blocks_sorted = sorted(
+            compute_blocks,
+            key=sort_blocks_key,
+            reverse=iteration_order == gt_ir.IterationOrder.BACKWARD,
+        )
+
+        # if sorting didn't change anything it was already sorted
+        return compute_blocks == compute_blocks_sorted
```

Comment:
Noticed when browsing this code that the `syntax_error` variable on line 607 is not defined.

Reply:
Good catch. Strange that mypy didn't catch this.